### PR TITLE
Task to restart failed deployements that are not from the dpl-cms project

### PR DESCRIPTION
**Pull request by ITViking**
#### What does this PR do?
This is hopefully a little nifty task that will make it faster and easier for us to redeploy failed deployments for both main and moduletest environments. 
I have done this as most deployment problems are solved by runnning them again. 
We should still investigate why deployments fail.

#### Should this be tested by the reviewer and how?
Just run it

#### Any specific requests for how the PR should be reviewed?
Read it and test it 

#### What are the relevant tickets?
https://reload.atlassian.net/jira/software/c/projects/DDFDRIFT/boards/464?selectedIssue=DDFDRIFT-252

### CODE REVIEW

File: infrastructure/Taskfile.yml [hunk 1]
```diff
@@ -1104,7 +1104,7 @@ tasks:
           lagoon raw --raw "query allProjects {
             allProjects {
             name
-              environments(type: PRODUCTION) {
+              environments(type: DEVELOPMENT) {
```
Comment (kasperg) [inline infrastructure/Taskfile.yml:1107]: Is this change intentional?

The comment for the task still states "Gets the latest deployment for each *production* environment".

In general I also think it is more valuable for us to see deployments to production over development.

File: infrastructure/Taskfile.yml [hunk 2]
```diff
@@ -1114,6 +1114,12 @@ tasks:
             }
           }" | jq -r '.allProjects[] | .name as $name | .environments[].deployments[] | select(.status != "complete") | ($name) + ": " + (.status)' 
 
+    sites:redeploy-failed-deployments:
+      deps: [lagoon:cli:config]
+      desc: redeployyes all failed deployments
```
Comment (kasperg) [inline infrastructure/Taskfile.yml:1119]:
```suggestion
      desc: Redeploys all failed deployments
```

File: infrastructure/task/scripts/redeploy-failed-deployments.sh [hunk 3]
```diff
@@ -0,0 +1,57 @@
+#!/usr/bin/env bash
+#
+# 80% of failed deployments is fixed by redeploying the failed deployment.
+# This script will help make that easier to do.
+
+echo "Will now start to redeployg failed deployments"
```
Comment (kasperg) [inline infrastructure/task/scripts/redeploy-failed-deployments.sh:6]:
```suggestion
echo "Will now start to redeploying failed deployments"
```

File: infrastructure/task/scripts/redeploy-failed-deployments.sh [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+      name
+      deployments(limit: 1) {
+        status
+        created
+      }
+    }
+  }
+}" | jq -r '.allProjects[] | .name as $name | .environments[].deployments[] | select(.status == "failed") | ($name)')
+
+echo "redeploying failed production environments"
```
Comment (kasperg) [inline infrastructure/task/scripts/redeploy-failed-deployments.sh:22]:
```suggestion
echo "Redeploying failed production environments"
```

File: infrastructure/task/scripts/redeploy-failed-deployments.sh [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+      name
+      deployments(limit: 1) {
+        status
+        created
+      }
+    }
+  }
+}" | jq -r '.allProjects[] | .name as $name | .environments[].deployments[] | select(.status == "failed") | ($name)')
+
+echo "redeploying failed moduletest environments"
```
Comment (kasperg) [inline infrastructure/task/scripts/redeploy-failed-deployments.sh:47]:
```suggestion
echo "Redeploying failed moduletest environments"
```

File: infrastructure/task/scripts/redeploy-failed-deployments.sh [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+
+for deployment in $FAILED_DEVELOPMENT_DEPLOYMENTS; do
+  if [[ $deployment =~ "dpl-cms" ]]; then
+    continue
+  fi
+  echo "$deployment: deploying"
+  lagoon deploy latest -p "$deployment" -e "moduletest" --force
+done
```
Comment (kasperg) [inline infrastructure/task/scripts/redeploy-failed-deployments.sh:55]: There is some repetition here that I wonder if we could refactor to reusable functions. I do not think it is that bad though. Keeping it simple is also nice.

**Review by kasperg — APPROVED** [review]
:+1: Nice.

I think you need to reconsider the change to `sites:incomplete-deployments` but I trust you can do this on your own.

I also have a few other suggestions and comments for you to consider.

Consider updating the deployment runbooks to refer to this task.

File: infrastructure/task/scripts/redeploy-failed-deployments.sh (hunk 6 above)
Comment (ITViking) [reply infrastructure/task/scripts/redeploy-failed-deployments.sh]: I considered that too when I wrote it. My conclusion was that this is a temporary solution to a problem we want to solve more completely, so for now let's just keep this simple
